### PR TITLE
[FLINK-24737][runtime-web] Update outdated web dependencies

### PR DESCRIPTION
**Pull request by yangjunhan**
## What is the purpose of the change

Update outdated web dependencies and dev-dependencies.
https://issues.apache.org/jira/browse/FLINK-24737

## Brief change log

Most dependencies have been updated to the latest versions except Angular (v13) and its dependent Rxjs and TypeScript, etc. According to `npm audit`, the remaining unresolved dependency vulnerabilities are mainly caused by Angular and should be fixed while bumping it to v13.

_Note: Bumping the Angular framework to v13 should be in another PR once NG-Zorro-Antd releases their compatible v13._

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-runtime-web/web-dashboard/package.json [hunk 1]
```diff
@@ -9,72 +9,73 @@
     "lint": "eslint --cache src --ext .ts,.html && stylelint \"**/*.less\"",
     "lint:fix": "eslint --fix --cache src --ext .ts,.html && stylelint \"**/*.less\" --fix",
     "ci-check": "npm run lint && npm run build",
-    "proxy": "ng serve --proxy-config proxy.conf.json"
+    "proxy": "ng serve --proxy-config proxy.conf.json",
+    "prepare": "cd ../.. && husky install flink-runtime-web/web-dashboard/.husky",
+    "lint-staged": "lint-staged"
   },
-  "private": true,
+  "private": false,
```
Comment (Airblader) [inline flink-runtime-web/web-dashboard/package.json:16]: What's the reason to change this? We don't ever want to publish this as a package.

Comment (yangjunhan) [reply flink-runtime-web/web-dashboard/package.json:16]: Ok, I will revert it.

**Comment by MartijnVisser** [comment]
@yangjunhan Can you have a look at the failing build and fix it?

**Comment by yangjunhan** [comment]
> Can you have a look at the failing build and fix it?

@MartijnVisser Conflicts have been addressed.

**Comment by MartijnVisser** [comment]
> > Can you have a look at the failing build and fix it?
> 
> @MartijnVisser Conflicts have been addressed.

Thanks @yangjunhan The CI unfortunately still fails and reports:

`[INFO] npm ERR! @jest/types not accessible from pretty-format`

**Comment by yangjunhan** [comment]
@flinkbot run azure

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Comment by MartijnVisser** [comment]
@Airblader Could you have a look?

**Review by Airblader — CHANGES_REQUESTED** [review]
Thanks for the PR and bringing the dependencies more up to date, @yangjunhan. Overall this looks good to me, but I have a couple questions.

However, it seems that now when making commits unrelated to the UI, there is output that didn't use to exist:

```
$ git commit …

> flink-dashboard@2.0.0 lint-staged
> lint-staged

→ No staged files match any configured task.
```

I think this will confuse the majority of developers as most do not work on the UI. Can we do something to restore the old behavior of not producing output unless the commit contains changes to the UI?

File: flink-runtime-web/web-dashboard/src/app/share/common/dagre/svg-container.component.ts [hunk 2]
```diff
@@ -73,20 +70,24 @@ export class SvgContainerComponent implements OnInit, AfterContentInit {
...
       .scaleExtent([this.nzMinZoom, this.nzMaxZoom])
-      .on('zoom', () => {
-        this.containerTransform = d3.event.transform;
-        this.zoom = this.containerTransform.k;
-        if (!isNaN(this.containerTransform.x)) {
-          this.transform = `translate(${this.containerTransform.x} ,${this.containerTransform.y})scale(${this.containerTransform.k})`;
+      .on('zoom', ({ transform }: SafeAny) => {
+        const { x, y, k } = transform;
+        this.zoom = k;
+        this.zoomEvent.emit(k);
```
[the remark this answers is not in the document]
Comment (yangjunhan) [reply flink-runtime-web/web-dashboard/src/app/share/common/dagre/svg-container.component.ts:82]: This seemed to be a mistake while resolving conflicts by rebasing. Let me fix it.

File: flink-runtime-web/web-dashboard/src/app/share/common/dagre/svg-container.component.ts [hunk 3]
```diff
@@ -73,20 +70,24 @@ export class SvgContainerComponent implements OnInit, AfterContentInit {
...
+        this.containerTransform = transform;
+        this.transformEvent.emit(transform);
```
[the remark this answers is not in the document]
Comment (yangjunhan) [reply flink-runtime-web/web-dashboard/src/app/share/common/dagre/svg-container.component.ts:84]: same as above

**Comment by yangjunhan** [comment]
> I think this will confuse the majority of developers as most do not work on the UI. Can we do something to restore the old behavior of not producing output unless the commit contains changes to the UI?

@Airblader  There is such output because "flink-runtime-web" is introducing **husky + lint-staged** for linting which essentially uses git hook through pre-commit, and the pre-commit is at the repo level.

One way I can think of to restore the older behavior is triggering npm script lint-staged only when `git diff --cached --quiet -- "flink-runtime-web/web-dashboard/*"` (maybe written in an independent shell script). Could you think of any other better way to achieve so?

**Comment by yangjunhan** [comment]
> @yangjunhan I apologize for the late response; I was on vacation over the holidays.
> 
> The changes overall look fine to me, except for the open point around the linting output. I think the solution you proposed sounds good, at least I can't think of anything better either.

@Airblader Hope you enjoyed your holidays. Since NG-ZORRO-antd recently released its new version, I am considering bumping Runtime-Web directly to Angular v13 and NG-ZORRO-antd v13 in this PR, if that is ok with you and @MartijnVisser .

**Comment by MartijnVisser** [comment]
@yangjunhan I'm not sure if it's better to make it a separate PR (since we also have https://issues.apache.org/jira/browse/FLINK-24768 open for this) or if we should do it in a separate commit. Let's see what @Airblader says on that. It would be great do have this update in place!

**Comment by Airblader** [comment]
I would prefer bumping this in a separate PR using the JIRA issue @MartijnVisser pointed to.

**Comment by yangjunhan** [comment]
> I would prefer bumping this in a separate PR using the JIRA issue @MartijnVisser pointed to. I'm happy to review and merge that one as well, of course (no vacation in between this time).

@Airblader No problems! I can do it in a separate PR. Meanwhile, I fixed the linting output problem as we discussed.

**Review by Airblader — APPROVED** [review]
Thanks!

**Pull request closed without merging**